qcodes/instrument_drivers/oxford/MercuryiPS_VISA.py

```diff
@@ -299,8 +299,89 @@ def __init__(self, name: str, address: str, visalib=None,
                            get_cmd=self._get_ramp_rate,
                            set_cmd=self._set_ramp_rate)
 
+
+        self.add_parameter('theta_saferamp',
+						   label='Theta',
+						   unit='degrees',
+						   set_cmd=self._theta_saferamp_set,
+						   get_cmd=self._theta_saferamp_get)
+		
+        self.add_parameter('r_saferamp',
+						   label='B_Radius',
+						   unit='T',
+						   set_cmd=self._r_saferamp_set,
+						   get_cmd=self._r_saferamp_get)
+
+        self.add_parameter('r_simulramp',
+						   label='B_Radius',
+						   unit='T',
+						   set_cmd=self._r_simulramp_set,
+						   get_cmd=self._r_simulramp_get)
+
+        self.add_parameter('phi_saferamp',
+						   label='Phi',
+						   unit='degrees',
+						   set_cmd=self._phi_saferamp_set,
+						   get_cmd=self._phi_saferamp_get)
+
         self.connect_message()
 
+    def _theta_saferamp_set(self, target):
+        """
+		Set the theta target and ramp to it
+		"""
+        self.theta_target(target)
+        self.ramp(mode='safe')
+        self.theta_measured()
+
+    def _theta_saferamp_get(self):
```

Comment:
what are these `get*` methods needed for if they just refer to the parameters which actually represent the values (like `theta_measured`)? I'd suggest to use set-only parameters for the ramping, and if values need to be read, then appropriate corresponding parameters are used. The reason for this is to remove redundancy if it brings extra maintainability and perhaps confusion.